lib/src/intl/string_iterator.dart

```diff
@@ -2,25 +2,41 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// Provides an 'iterator' over a string as a list of substrings, and also
-/// gives us a lookahead of one via the [peek] method.
+import 'dart:math';
+
+/// An indexed position in a String which can read by specified character
+/// counts, or read digits up to a delimiter.
 class StringIterator {
```

Comment:
nit: Within the sdk we generally use a name like `FooIterator` only for objects that implement the `Iterator` interface.

Reply:
I agree!